python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -365,6 +368,19 @@ def _check_getter(self):
             raise AttributeError('cannot modify or delete "%s" property (declared as "getter")' % _getName(self))
 
     def __get__(self, obj, cls):
+
+        self_name = _getName(self)
+        GangaObject._acquireChildLock(self_name, obj)
+
+        try:
+            returnable = self.__really_get(obj, cls)
+        finally:
+            GangaObject._releaseChildLock(self_name, obj)
+
+        return returnable
+
+    def __really_get(self, obj, cls):
```

Comment:
I'd rather avoid all these levels of indiretion in the function calls. Either we should just lock around the contents of `__get__` with a context manager or put together a small decorator which locks for a function.